packages/dts-generator/src/utils/json-constructor-settings-interfaces.ts

```diff
@@ -156,8 +156,8 @@ function createConstructorSettingsInterfaces(
 
           if (
             !(
-              prop.type.kind === "TypeReference" &&
-              prop.type.typeName === "string"
+              prop.valueType.kind === "TypeReference" &&
```

Comment:
I'm not happy with the name `valueType`. The old `type` also is a type of the property's value. It's just a broader, less limiting type. 

The new type is more specific, it's the semantic type whereas `type` is the type that ManagedObject can handle.

Reply:
`formalType` (as opposed to "actual" runtime type?)
or
`apiType` since it's the APIs signature?